exhibitor-standalone/src/main/java/com/netflix/exhibitor/application/ExhibitorMain.java

```diff
@@ -131,20 +131,42 @@ public ExhibitorMain(BackupProvider backupProvider, ConfigProvider configProvide
         ServletContainer        container = new ServletContainer(application);
 
         server = new Server();
+        HttpsConfiguration httpsConf = exhibitor.getHttpsConfiguration();
 
-        // By default the jetty.bio.SocketConnector is used. The
-        // SocketConnector performs blocking I/O and so suffers from
-        // spawning a new thread per connection. To improve
-        // performance and limit the number of threads we switch to
-        // the jetty.nio.SelectChannelConnector. This is a
-        // non-blocking I/O connector.
-        // See https://dcosjira.atlassian.net/browse/DCOS-558
-        SelectChannelConnector connector = new SelectChannelConnector();
-        connector.setPort(httpPort);
-        connector.setAcceptors(8);
-        connector.setMaxIdleTime(5000);
-        connector.setAcceptQueueSize(32);
-        server.setConnectors(new Connector[]{connector});
+        if ( httpsConf.getServerKeystorePath() != null )
+        {
+            SslSocketConnector connector = new SslSocketConnector();
```

Comment:
If I remember correctly, the SslSocketConnector suffers from the same issues described in the comment below relating to the SocketConnector (ie., it has a blockling-IO implementation). Please use `SslSelectChannelConnector` and configure it with the same settings used below:
```
+            connector.setAcceptors(8);
 +            connector.setMaxIdleTime(5000);
 +            connector.setAcceptQueueSize(32);
```